systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -216,6 +223,112 @@ void testUserWithQuotas(KafkaUser user) {
+                        .withNewSecretPrefix(secretPrefix)
+                    .endUserOperator()
+                .endEntityOperator()
+            .endSpec()
+            .done();
+
+        KafkaTopicResource.topic(clusterName, TOPIC_NAME).done();
+        KafkaUser tlsUser = KafkaUserResource.tlsUser(clusterName, tlsUserName).done();
+        KafkaUser scramShaUser = KafkaUserResource.scramShaUser(clusterName, scramShaUserName).done();
+
+        LOGGER.info("Deploying KafkaClients pod for TLS listener");
+        KafkaClientsResource.deployKafkaClients(true, clusterName + "-tls-" + Constants.KAFKA_CLIENTS, true, Constants.TLS_LISTENER_DEFAULT_NAME, secretPrefix, tlsUser).done();
+        String tlsKafkaClientsName = kubeClient().listPodsByPrefixInName(clusterName + "-tls-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        LOGGER.info("Deploying KafkaClients pod for PLAIN listener");
+        KafkaClientsResource.deployKafkaClients(false, clusterName + "-plain-" + Constants.KAFKA_CLIENTS, true, Constants.PLAIN_LISTENER_DEFAULT_NAME, secretPrefix, scramShaUser).done();
+        String plainKafkaClientsName = kubeClient().listPodsByPrefixInName(clusterName + "-plain-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(tlsKafkaClientsName)
+            .withNamespaceName(NAMESPACE)
+            .withTopicName(TOPIC_NAME)
+            .withKafkaUsername(tlsUserName)
+            .withSecurityProtocol(SecurityProtocol.SASL_SSL)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .withClusterName(clusterName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withSecretPrefix(secretPrefix)
+            .build();
+
+        LOGGER.info("Checking if TLS user is able to send messages");
+        internalKafkaClient.assertSentAndReceivedMessages(
+            internalKafkaClient.sendMessagesTls(),
+            internalKafkaClient.receiveMessagesTls()
+        );
+
+        internalKafkaClient = internalKafkaClient.toBuilder()
+            .withUsingPodName(plainKafkaClientsName)
+            .withSecurityProtocol(SecurityProtocol.PLAINTEXT)
+            .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+            .withKafkaUsername(scramShaUserName)
+            .build();
+
+        LOGGER.info("Checking if SCRAM-SHA user is able to send messages");
+        internalKafkaClient.assertSentAndReceivedMessages(
+            internalKafkaClient.sendMessagesPlain(),
+            internalKafkaClient.receiveMessagesPlain()
+        );
+
+        Secret tlsSecret = kubeClient().getSecret(secretPrefix + tlsUserName);
+        Secret scramShaSecret = kubeClient().getSecret(secretPrefix + scramShaUserName);
+
+        LOGGER.info("Checking if user secrets with secret prefixes exists");
+        assertNotNull(tlsSecret);
+        assertNotNull(scramShaSecret);
```

Comment:
Should this be checked before sending the messages? It seems to me like a pre-requisite for it. Withotu the secrets, the clients should not work.